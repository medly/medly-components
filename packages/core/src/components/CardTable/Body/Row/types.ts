import { WithThemeProp } from '@medly-components/utils';

export interface Props extends WithThemeProp {
    onClick?: () => void;
    isRowClickDisabled?: boolean;
    withWhiteBackground?: boolean;
    gridTemplateColumns: string;
}
