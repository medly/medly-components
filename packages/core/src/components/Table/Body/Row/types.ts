import { HTMLProps, Omit, WithThemeProp } from '@medly-components/utils';
import { GridTemplateProps } from '../../types';

export type StyledProps = GridTemplateProps &
    HTMLProps<HTMLTableRowElement> &
    WithThemeProp & {
        isSelected?: boolean;
        isExpanded?: boolean;
        isExpandable?: boolean;
        showRowWithCardStyle?: boolean;
        withRowSeparators?: boolean;
    };
export type RowProps = Omit<HTMLProps<HTMLTableRowElement>, 'style' | 'data'> & {
    data: { [key: string]: any };
    selectedRowIds: Array<any>;
    onRowSelection: (id: any) => void;
    showShadowAfterFrozenElement?: boolean;
};
