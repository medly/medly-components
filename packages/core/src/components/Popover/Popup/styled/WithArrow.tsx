import { css } from 'styled-components';
import { PopoverPopupProps } from '../types';

const getPositionWithArrow = ({ placement }: PopoverPopupProps) => {
    switch (placement) {
        case 'top':
            return css`
                box-shadow: 0 0.4rem 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: translateX(-50%);
                    top: 100%;
                    left: 50%;
                }
            `;
        case 'top-start':
            return css`
                box-shadow: 0 0.4rem 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    top: 100%;
                    left: 2rem;
                }
            `;
        case 'top-end':
            return css`
                box-shadow: 0 0.4rem 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    top: 100%;
                    right: 2rem;
                }
            `;
        case 'right':
            return css`
                box-shadow: -0.4rem 0 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: translateY(-50%) rotate(90deg);
                    top: 50%;
                    right: calc(100% - 0.3rem);
                }
            `;
        case 'right-start':
            return css`
                box-shadow: -0.4rem 0 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: rotate(90deg);
                    top: 2rem;
                    right: calc(100% - 0.3rem);
                }
            `;
        case 'right-end':
            return css`
                box-shadow: -0.4rem 0 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: rotate(90deg);
                    bottom: 2rem;
                    right: calc(100% - 0.3rem);
                }
            `;
        case 'bottom':
            return css`
                box-shadow: 0 -0.4rem 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: translateX(-50%) rotate(180deg);
                    left: 50%;
                    bottom: 100%;
                }
            `;
        case 'bottom-start':
            return css`
                box-shadow: 0 -0.4rem 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: rotate(180deg);
                    left: 2rem;
                    bottom: 100%;
                }
            `;
        case 'bottom-end':
            return css`
                box-shadow: 0 -0.4rem 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: rotate(180deg);
                    right: 2rem;
                    bottom: 100%;
                }
            `;
        case 'left':
            return css`
                box-shadow: 0.4rem 0 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: translateY(-50%) rotate(270deg);
                    top: 50%;
                    left: calc(100% - 0.3rem);
                }
            `;
        case 'left-start':
            return css`
                box-shadow: 0.4rem 0 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: rotate(270deg);
                    top: 2rem;
                    left: calc(100% - 0.3rem);
                }
            `;
        case 'left-end':
            return css`
                box-shadow: 0.4rem 0 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: rotate(270deg);
                    bottom: 2rem;
                    left: calc(100% - 0.3rem);
                }
            `;
    }
};

export const getStyleWithArrow = css<PopoverPopupProps>`
    border-radius: 0.8rem;
    padding: 1.5rem 2rem;

    &::after {
        content: '';
        width: 0;
        height: 0;
        z-index: 4;
        position: absolute;
        border-left: 0.8rem solid transparent;
        border-right: 0.8rem solid transparent;
        border-top: 1rem solid ${({ bgColor }) => bgColor};
    }

    ${getPositionWithArrow}
`;
