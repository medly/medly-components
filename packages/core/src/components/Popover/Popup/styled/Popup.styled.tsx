import { fullHeight, fullWidth } from '@medly-components/utils';
import styled, { css } from 'styled-components';
import { Placement } from '../../types';
import { StyledPopoverPopupProps } from '../types';
import { getStyleWithArrow } from './WithArrow';

export const getPosition = ({ placement, distance = '0px' }: { placement: Placement; distance?: string }) => {
    switch (placement) {
        case 'top-start':
            return css`
                bottom: calc(100% + ${distance});
                left: 0%;
            `;
        case 'top':
            return css`
                bottom: calc(100% + ${distance});
                left: 50%;
                transform: translateX(-50%);
            `;
        case 'top-end':
            return css`
                bottom: calc(100% + ${distance});
                right: 0%;
            `;
        case 'right-start':
            return css`
                top: 0%;
                left: calc(100% + ${distance});
            `;
        case 'right':
            return css`
                left: calc(100% + ${distance});
                top: 50%;
                transform: translateY(-50%);
            `;
        case 'right-end':
            return css`
                bottom: 0;
                left: calc(100% + ${distance});
            `;
        case 'bottom-start':
            return css`
                top: calc(100% + ${distance});
                left: 0%;
            `;
        case 'bottom':
            return css`
                top: calc(100% + ${distance});
                left: 50%;
                transform: translateX(-50%);
            `;
        case 'bottom-end':
            return css`
                top: calc(100% + ${distance});
                right: 0%;
            `;
        case 'left-start':
            return css`
                right: calc(100% + ${distance});
                top: 0%;
            `;
        case 'left':
            return css`
                right: calc(100% + ${distance});
                top: 50%;
                transform: translateY(-50%);
            `;
        case 'left-end':
            return css`
                right: calc(100% + ${distance});
                bottom: 0%;
            `;
    }
};

export const PopupStyled = styled('div')<StyledPopoverPopupProps>`
    z-index: 4;
    position: absolute;
    background-color: ${({ bgColor }) => bgColor};

    ${props => props.fullWidth && fullWidth()};
    ${props => props.fullHeight && fullHeight()};

    ${props => getPosition({ placement: props.placement, distance: props.withArrow ? '1.5rem' : props.distance })}
    ${({ withArrow }) => withArrow && getStyleWithArrow}
`;
