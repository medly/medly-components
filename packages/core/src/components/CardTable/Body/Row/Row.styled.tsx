import styled, { css } from 'styled-components';
import { Props } from './types';

const getStyle = ({ withWhiteBackground, isRowClickDisabled }: Props) => {
    const bgTheme = withWhiteBackground ? 'lightBackground' : 'darkBackground';
    return css`
        border: 0.1rem solid ${({ theme }) => theme.cardTable[bgTheme].borderColor};
        box-shadow: ${({ theme }) => theme.cardTable[bgTheme].boxShadow};

        &:hover {
            box-shadow: ${({ theme }) => theme.cardTable[bgTheme].hoverBoxShadow};
        }

        :nth-child(odd) {
            background: ${({ theme }) =>
                !isRowClickDisabled ? theme.cardTable[bgTheme].oddRowBgColor : theme.cardTable.disabledRowBgColor};
        }
        :nth-child(even) {
            background: ${({ theme }) =>
                !isRowClickDisabled ? theme.cardTable[bgTheme].evenRowBgColor : theme.cardTable.disabledRowBgColor};
        }
    `;
};

export const Row = styled('tr')<Props>`
    display: grid;
    grid-template-columns: ${({ gridTemplateColumns }) => gridTemplateColumns};
    cursor: ${({ onClick, isRowClickDisabled }) => (onClick && !isRowClickDisabled ? 'pointer' : 'inherit')};
    box-sizing: border-box;
    border-radius: 0.8rem;
    margin-bottom: 0.8rem;
    background-color: white;
    transition: box-shadow 100ms ease-out;
    ${getStyle}
`;

export const NoResult = styled(Row)`
    display: flex;
    justify-content: center;
    align-items: center;
    padding: 1.5rem 0;
`;
