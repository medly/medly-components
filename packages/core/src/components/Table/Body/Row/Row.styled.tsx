import { rgba } from 'polished';
import styled, { css } from 'styled-components';
import { GridTemplateProps } from '../../types';
import { defaultTableCellPaddings } from '../Cell/Styled';
import { StyledProps } from './types';

const getRowBackgroundColor = ({ theme, isSelected, disabled, even }: StyledProps & { even: boolean }) => {
    const parity = even ? 'even' : 'odd';
    return theme.table.row.bgColor[disabled ? 'disabled' : isSelected ? 'selected' : parity];
};

const getHoverStyleForCardStyle = css`
    box-shadow: ${({ theme }) => `0 0.2rem 0.8rem ${rgba(theme.table.shadowColor, 0.35)} `};
`;

const cardStyle = css<StyledProps>`
    border: none;
    border-radius: 0.8rem;
    margin-left: 0.8rem;
    margin-right: 0.8rem;
    box-shadow: ${({ theme }) => `0 0.2rem 0.8rem ${rgba(theme.table.shadowColor, 0.2)} `};
    &,
    & > * {
        background-color: ${({ theme, isSelected, disabled }) => getRowBackgroundColor({ theme, isSelected, disabled, even: true })};
        color: ${({ theme, isSelected, disabled }) => theme.table.row.textColor[disabled ? 'disabled' : isSelected ? 'selected' : 'even']};
    }

    &&& > * {
        &:first-child {
            &,
            &::before {
                transition: ${props => !props.isExpanded && 'border-radius 200ms ease-in'};
                border-radius: ${props => (props.isExpanded ? '0.8rem 0 0 0' : '0.8rem 0 0 0.8rem')};
            }

            &::before {
                width: 0.8rem;
            }
        }
        &:last-child {
            border-radius: ${props => (props.isExpandable ? '0 0 0.8rem 0' : '0 0.8rem 0.8rem 0')};
        }
        &:nth-last-child(3) {
            border-radius: ${props => props.isExpandable && (props.isExpanded ? '0 0.8rem 0 0' : '0 0.8rem 0.8rem 0')};
        }
        &:nth-last-child(2) {
            border-radius: ${props => props.isExpandable && '0 0 0 0.8rem'};
            &::before {
                width: 0.8rem;
                border-radius: 0 0 0 0.8rem;
            }
        }
    }

    &:hover {
        box-shadow: ${({ theme }) => `0 0.2rem 0.8rem ${rgba(theme.table.shadowColor, 0.35)} `};
    }

    &:not(:last-child) {
        margin-bottom: 0.8rem;
    }
`;

const getHoverStyleForNormalStyle = (style: 'shadow' | 'outlined') =>
    style === 'outlined'
        ? css<StyledProps>`
              z-index: 2;
              border-radius: 0.2rem;
              border: 0.2rem solid ${({ theme }) => theme.table.row.hoveredStyle.color};
              padding-bottom: 0;
          `
        : css<StyledProps>`
              z-index: 2;
              box-shadow: ${({ disabled, onClick, theme }) =>
                  !disabled && onClick && `0 0.2rem 0.4rem ${rgba(theme.table.shadowColor, 0.2)} `};
          `;

const getBorderStyle = (rowHoveredStyle: 'shadow' | 'outlined') =>
    rowHoveredStyle === 'outlined'
        ? css<StyledProps>`
              border-width: 0.2rem;
              border-style: solid;

              &:nth-child(odd) {
                  border-color: ${({ theme, isSelected, disabled }) => getRowBackgroundColor({ theme, isSelected, disabled, even: false })};
              }

              &:nth-child(even) {
                  border-color: ${({ theme, isSelected, disabled }) => getRowBackgroundColor({ theme, isSelected, disabled, even: true })};
              }
          `
        : ``;

const getPadding = (rowHoveredStyle: 'shadow' | 'outlined') =>
    rowHoveredStyle === 'outlined'
        ? css`
              padding-bottom: 0.1rem;
          `
        : '';

const rowSeparatorStyle = css<StyledProps>`
    &:not(:last-child) {
        border-bottom: 0.1rem solid ${({ theme }) => theme.table.row.separatorColor};
        ${({ theme }) => getPadding(theme.table.row.hoveredStyle.style)}
    }
`;

const normalStyle = css<StyledProps>`
    &&:hover {
        ${({ theme }) => getHoverStyleForNormalStyle(theme.table.row.hoveredStyle.style)};
    }

    &:nth-child(odd) {
        &,
        & > * {
            background-color: ${({ theme, isSelected, disabled }) => getRowBackgroundColor({ theme, isSelected, disabled, even: false })};
            color: ${({ theme, isSelected, disabled }) =>
                theme.table.row.textColor[disabled ? 'disabled' : isSelected ? 'selected' : 'odd']};
        }
    }

    &:nth-child(even) {
        &,
        & > * {
            background-color: ${({ theme, isSelected, disabled }) => getRowBackgroundColor({ theme, isSelected, disabled, even: true })};
            color: ${({ theme, isSelected, disabled }) =>
                theme.table.row.textColor[disabled ? 'disabled' : isSelected ? 'selected' : 'even']};
        }
    }

    ${({ theme }) => getBorderStyle(theme.table.row.hoveredStyle.style)};
    ${({ withRowSeparators }) => withRowSeparators && rowSeparatorStyle}
`;

export const Row = styled('tr').attrs(({ gridTemplateColumns }: GridTemplateProps) => ({
    style: {
        gridTemplateColumns
    }
}))<StyledProps>`
    display: grid;
    position: relative;
    align-items: center;
    min-width: fit-content;
    outline: none;
    cursor: ${({ disabled, onClick }) => (disabled ? 'not-allowed' : onClick ? 'pointer' : 'inherit')};
    ${({ showRowWithCardStyle }) => (showRowWithCardStyle ? cardStyle : normalStyle)}
    ${({ isNavigated, showRowWithCardStyle, theme }) =>
        isNavigated && (showRowWithCardStyle ? getHoverStyleForCardStyle : getHoverStyleForNormalStyle(theme.table.row.hoveredStyle.style))}
`;

export const NoResultCell = styled('td')<{ width: number; tableSize: keyof typeof defaultTableCellPaddings }>`
    text-align: center;
    padding: ${({ tableSize }) => defaultTableCellPaddings[tableSize]};
    width: ${({ width }) => `${width}px`};
`;

export const NoResultRow = styled(Row)<{ withMinimap?: boolean }>`
    justify-content: center;
    align-items: center;
    ${({ withMinimap }) => withMinimap && `min-height: 4.6rem;`}
`;
