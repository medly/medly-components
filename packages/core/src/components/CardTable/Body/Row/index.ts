import { Row } from './Row.styled';

export default Row;
