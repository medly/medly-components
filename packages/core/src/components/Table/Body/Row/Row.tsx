import type { FC, KeyboardEvent } from 'react';
import { memo, useCallback, useContext, useEffect, useMemo, useRef, useState } from 'react';
import { TableComponentsCommonPropsContext } from '../../context';
import { GroupCell } from '../../GroupCell';
import { getGridTemplateColumns, getNestedValue } from '../../helpers';
import { TableColumnConfig } from '../../types';
import Cell from '../Cell';
import ExtendedRowCell from '../Cell/ExtendedRowCell';
import RowActionsCell from '../Cell/RowActionsCell';
import RowHoverActionsCell from '../Cell/RowHoverActionsCell';
import * as Styled from './Row.styled';
import { RowProps } from './types';

export const Row: FC<RowProps> = memo(props => {
    const [isRowHovered, setIsRowHovered] = useState(false),
        {
            id,
            data,
            isNavigated = false,
            isRowExpandedFromKeyboard,
            isRowCollapsedFromKeyboard,
            showShadowAfterFrozenElement,
            selectedRowIds,
            onRowSelection,
            ...restProps
        } = props,
        {
            keyBindings,
            columns,
            isLoading,
            onRowClick,
            hiddenDivRef,
            addColumnMaxSize,
            rowClickDisableKey,
            rowSelectionDisableKey,
            defaultRowExpandKey,
            isRowSelectable,
            isRowExpandable,
            isGroupedTable,
            showRowWithCardStyle,
            size: tableSize,
            expandedRowComponent,
            withRowSeparators,
            rowHoverActions: RowHoverActions
        } = useContext(TableComponentsCommonPropsContext);

    const [isExpanded, setExpansionState] = useState<boolean>(defaultRowExpandKey && data[defaultRowExpandKey]),
        isRowSelected = useMemo(() => !isLoading && selectedRowIds.includes(id), [id, isLoading, selectedRowIds]),
        isRowClickDisabled = useMemo(() => rowClickDisableKey && data[rowClickDisableKey], [data, rowClickDisableKey]),
        isRowSelectionDisabled = useMemo(() => rowSelectionDisableKey && data[rowSelectionDisableKey], [data, rowSelectionDisableKey]),
        handleRowSelection = useCallback(() => onRowSelection(id), [id, onRowSelection]),
        handleExpansionIconClick = useCallback(() => setExpansionState(val => !val), []),
        handleRowClick = useMemo(
            () => (!isLoading && onRowClick && !isRowClickDisabled ? () => onRowClick(data) : undefined),
            [isLoading, data, onRowClick, isRowClickDisabled, isRowExpandable, handleExpansionIconClick]
        ),
        handleMouseEnter = useCallback(() => setIsRowHovered(true), []),
        handleMouseLeave = useCallback(() => setIsRowHovered(false), []),
        ref = useRef<HTMLTableRowElement>(null);

    const getCells = useCallback(
        (rowData: any = {}, configs: TableColumnConfig[] = columns, field = '') =>
            configs.reduce((cells, config) => {
                const fieldName = `${field && `${field}.`}${config.field}`;

                return config.field === 'row-actions'
                    ? cells
                    : [
                          ...cells,
                          config.children ? (
                              <GroupCell
                                  as={field ? 'div' : 'td'}
                                  key={config.field}
                                  hidden={config.hidden}
                                  gridTemplateColumns={getGridTemplateColumns(config.children)}
                              >
                                  {getCells(rowData[config.field], config.children, config.field)}
                              </GroupCell>
                          ) : (
                              <Cell
                                  rowId={id}
                                  key={fieldName}
                                  config={config}
                                  as={field ? 'div' : 'td'}
                                  isLoading={isLoading}
                                  rowData={rowData}
                                  data={getNestedValue(rowData, config.field)}
                                  addColumnMaxSize={addColumnMaxSize}
                                  isRowClickDisabled={isRowClickDisabled}
                                  dottedFieldName={fieldName}
                                  tableSize={tableSize}
                                  hiddenDivRef={hiddenDivRef}
                              />
                          )
                      ];
            }, [] as React.ReactElement[]),
        [id, data, isLoading, columns, addColumnMaxSize, isRowClickDisabled, hiddenDivRef, tableSize]
    );

    useEffect(() => {
        if (isNavigated) {
            if (isRowSelectable && !isRowSelectionDisabled) ref.current?.querySelector('input')?.focus();
            else ref.current?.focus();
        }
    }, [isNavigated]);

    useEffect(() => {
        isRowExpandedFromKeyboard && setExpansionState(true);
    }, [isRowExpandedFromKeyboard]);

    useEffect(() => {
        isRowCollapsedFromKeyboard && setExpansionState(false);
    }, [isRowCollapsedFromKeyboard]);

    // TODO: Check why useKeypress is not working in this case
    const handleRowClickFromKeyboard = useCallback(
        (e: KeyboardEvent<HTMLTableRowElement>) => {
            e.key === keyBindings.rowClick! && isNavigated && handleRowClick && handleRowClick();
        },
        [isNavigated, handleRowClick]
    );

    return (
        <>
            <Styled.Row
                {...restProps}
                ref={ref}
                tabIndex={Number(id)}
                disabled={isRowClickDisabled}
                onClick={handleRowClick}
                isSelected={isRowSelected}
                isExpanded={isExpanded}
                isExpandable={isRowExpandable}
                showRowWithCardStyle={showRowWithCardStyle}
                gridTemplateColumns={getGridTemplateColumns(columns)}
                onMouseEnter={handleMouseEnter}
                onMouseLeave={handleMouseLeave}
                withRowSeparators={withRowSeparators}
                isNavigated={isNavigated}
                onKeyDown={handleRowClickFromKeyboard}
            >
                {(isRowSelectable || isRowExpandable) && (
                    <RowActionsCell
                        tableSize={tableSize}
                        isLoading={isLoading}
                        isRowExpanded={isExpanded}
                        isGroupedTable={isGroupedTable}
                        isRowExpandable={isRowExpandable}
                        isRowSelected={isRowSelected}
                        isRowSelectable={isRowSelectable}
                        isRowSelectionDisabled={isRowSelectionDisabled}
                        onRowSelection={handleRowSelection}
                        onRowExpansionIconClick={handleExpansionIconClick}
                        showShadowAtRight={showShadowAfterFrozenElement}
                        isNavigated={isNavigated}
                    />
                )}
                {getCells(data)}
                {isRowExpandable && expandedRowComponent && (
                    <ExtendedRowCell
                        rowId={id!}
                        rowData={data}
                        tableSize={tableSize}
                        isRowExpanded={isExpanded}
                        isRowSelected={isRowSelected}
                        isGroupedTable={isGroupedTable}
                        expandedRowComponent={expandedRowComponent}
                        isRowClickDisabled={isRowClickDisabled}
                        showShadowAtRight={showShadowAfterFrozenElement}
                    />
                )}
                {RowHoverActions !== undefined && !isRowClickDisabled && !isLoading && (
                    <RowHoverActionsCell show={isRowHovered}>
                        <RowHoverActions rowData={data} rowId={id} />
                    </RowHoverActionsCell>
                )}
            </Styled.Row>
        </>
    );
});
Row.displayName = 'Row';
