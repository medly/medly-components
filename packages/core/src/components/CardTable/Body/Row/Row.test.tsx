import { fireEvent, render } from '@test-utils';
import { Row } from './Row.styled';

const renderer = ({ withWhiteBackground = false, onClick = jest.fn(), isRowClickDisabled = false }) =>
    render(
        <table>
            <tbody>
                <Row
                    withWhiteBackground={withWhiteBackground}
                    onClick={onClick}
                    isRowClickDisabled={isRowClickDisabled}
                    gridTemplateColumns="1fr"
                >
                    <td>Demo</td>
                </Row>
            </tbody>
        </table>
    );

describe('CardTable Row', () => {
    it('should render properly with dark theme', () => {
        const { container } = renderer({ withWhiteBackground: false });
        expect(container).toMatchSnapshot();
    });
    it('should render properly with light theme', () => {
        const { container } = renderer({ withWhiteBackground: true });
        expect(container).toMatchSnapshot();
    });
    it('should render with grey background if row is disabled', () => {
        const { container } = renderer({ isRowClickDisabled: true });
        expect(container).toMatchSnapshot();
    });
    it('should call onClick on click of the row', () => {
        const onClick = jest.fn();
        const { getByText } = renderer({ isRowClickDisabled: false, onClick });
        fireEvent.click(getByText('Demo'));
        expect(onClick).toHaveBeenCalled();
    });
    it('should not call onClick on click of the row, if the row is disabled', () => {
        const onClick = jest.fn();
        const { getByText } = renderer({ isRowClickDisabled: true, onClick: undefined });
        fireEvent.click(getByText('Demo'));
        expect(onClick).not.toHaveBeenCalled();
    });
});
