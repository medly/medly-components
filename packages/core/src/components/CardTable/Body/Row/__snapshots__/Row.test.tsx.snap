// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardTable Row should render properly with dark theme 1`] = `
.c0 {
  display: grid;
  grid-template-columns: 1fr;
  cursor: pointer;
  box-sizing: border-box;
  border-radius: 0.8rem;
  margin-bottom: 0.8rem;
  background-color: white;
  transition: box-shadow 100ms ease-out;
  border: 0.1rem solid #ffffff;
  box-shadow: 0 0.2rem 0.8rem rgba(176,188,200,0.2);
}

.c0:hover {
  box-shadow: 0 0.4rem 0.8rem rgba(176, 188, 200, 0.6);
}

.c0 :nth-child(odd) {
  background: #ffffff;
}

.c0 :nth-child(even) {
  background: #ffffff;
}

<div>
  <table>
    <tbody>
      <tr
        class="c0"
        gridtemplatecolumns="1fr"
      >
        <td>
          Demo
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`CardTable Row should render properly with light theme 1`] = `
.c0 {
  display: grid;
  grid-template-columns: 1fr;
  cursor: pointer;
  box-sizing: border-box;
  border-radius: 0.8rem;
  margin-bottom: 0.8rem;
  background-color: white;
  transition: box-shadow 100ms ease-out;
  border: 0.1rem solid #dfe4e9;
  box-shadow: none;
}

.c0:hover {
  box-shadow: 0px 0.2rem 0.4rem rgba(176, 188, 200, 0.6);
}

.c0 :nth-child(odd) {
  background: #F7F8F9;
}

.c0 :nth-child(even) {
  background: #ffffff;
}

<div>
  <table>
    <tbody>
      <tr
        class="c0"
        gridtemplatecolumns="1fr"
      >
        <td>
          Demo
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`CardTable Row should render with grey background if row is disabled 1`] = `
.c0 {
  display: grid;
  grid-template-columns: 1fr;
  cursor: inherit;
  box-sizing: border-box;
  border-radius: 0.8rem;
  margin-bottom: 0.8rem;
  background-color: white;
  transition: box-shadow 100ms ease-out;
  border: 0.1rem solid #ffffff;
  box-shadow: 0 0.2rem 0.8rem rgba(176,188,200,0.2);
}

.c0:hover {
  box-shadow: 0 0.4rem 0.8rem rgba(176, 188, 200, 0.6);
}

.c0 :nth-child(odd) {
  background: #eff2f4;
}

.c0 :nth-child(even) {
  background: #eff2f4;
}

<div>
  <table>
    <tbody>
      <tr
        class="c0"
        gridtemplatecolumns="1fr"
      >
        <td>
          Demo
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
